app/queues/certificate/order-completer-worker.server.ts

```diff
@@ -1,19 +1,91 @@
-import { Worker } from 'bullmq';
+import { Worker, UnrecoverableError } from 'bullmq';
+import { CertificateStatus } from '@prisma/client';
 import { redis } from '~/lib/redis.server';
 import logger from '~/lib/logger.server';
+import LetsEncrypt from '~/lib/lets-encrypt.server';
+import * as certificateModel from '~/models/certificate.server';
 
 export interface OrderCompleterData {
   rootDomain: string;
   username: string;
+  certificateId: number;
 }
 
 export const orderCompleterQueueName = 'certificate-completeOrder';
 
 export const orderCompleterWorker = new Worker<OrderCompleterData>(
   orderCompleterQueueName,
   async (job) => {
-    const { rootDomain } = job.data;
-    logger.info(`TODO complete order for ${rootDomain}`);
+    const { rootDomain, certificateId } = job.data;
+
+    logger.info('Attempting to complete ACME order with the provider', {
+      rootDomain,
+      certificateId,
+    });
+
+    /**
+     * Reload certificate URL from DB
+     */
+    let certificateEntry;
+    try {
+      certificateEntry = await certificateModel.getCertificateById(certificateId);
+    } catch (e) {
+      // If we cannot recall the record from db, it's certainly an unrecoverable error
+      logger.error(
+        `Failed to reload certificate ${certificateId} from DB, rethrowing error as Unrecoverable`,
+        e
+      );
+
+      const newError = new UnrecoverableError((e as Error).message);
+      newError.stack = (e as Error).stack;
+      throw newError;
+    }
+
+    /**
+     * Initialize Let's Encrypt
+     */
+    let letsEncrypt;
+    try {
+      logger.debug("Initializing Let's encrypt");
```

Comment:
Why does this file change to using " vs. '?  I'm surprised Prettier isn't mad.

Reply:
It's because there is an apostrophe in the string. If that wasn't there, it would switch it to `'` This way there is no need to escape that char